sdk/loadtestservice/Azure.Developer.LoadTesting/src/LoadTestRunClient.cs

```diff
@@ -90,5 +90,67 @@ public virtual async Task<TestRunOperation> BeginTestRunAsync(WaitUntil waitUnti
+            return PageableHelpers.CreateAsyncPageable(FirstPageRequest, NextPageRequest, e => BinaryData.FromString(e.GetRawText()), ClientDiagnostics, _pipeline, "LoadTestRunClient.GetMetrics", "value", "nextLink", context);
+        }
+
+        /// <summary> List the metric values for a load test run. </summary>
+        /// <param name="testRunId"> Unique name for the load test run, must contain only lower-case alphabetic, numeric, underscore or hyphen characters. </param>
+        /// <param name="metricname"> Metric name. </param>
+        /// <param name="metricNamespace"> Metric namespace to query metric definitions for. </param>
+        /// <param name="timespan"> The timespan of the query. It is a string with the following format &apos;startDateTime_ISO/endDateTime_ISO&apos;. </param>
+        /// <param name="content"> The content to send as the body of the request. Details of the request body schema are in the Remarks section below. </param>
+        /// <param name="aggregation"> The aggregation. </param>
+        /// <param name="interval"> The interval (i.e. timegrain) of the query. Allowed values: &quot;PT5S&quot; | &quot;PT10S&quot; | &quot;PT1M&quot; | &quot;PT5M&quot; | &quot;PT1H&quot;. </param>
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="testRunId"/>, <paramref name="metricname"/>, <paramref name="metricNamespace"/> or <paramref name="timespan"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="testRunId"/> is an empty string, and was expected to be non-empty. </exception>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The <see cref="Pageable{T}"/> from the service containing a list of <see cref="BinaryData"/> objects. Details of the body schema for each item in the collection are in the Remarks section below. </returns>
+        /// <include file="Generated/Docs/LoadTestRunClient.xml" path="doc/members/member[@name='GetMetrics(String,String,String,String,RequestContent,String,String,RequestContext)']/*" />
+        public virtual Pageable<BinaryData> GetMetrics(string testRunId, string metricname, string metricNamespace, string timespan, RequestContent content = null, string aggregation = null, string interval = null, RequestContext context = null)
+        {
+            Argument.AssertNotNullOrEmpty(testRunId, nameof(testRunId));
+            Argument.AssertNotNull(metricname, nameof(metricname));
+            Argument.AssertNotNull(metricNamespace, nameof(metricNamespace));
+            Argument.AssertNotNull(timespan, nameof(timespan));
+
+            if (content == null)
+            {
+                content = RequestContent.Create(new { });
+            }
```

Comment:
It seems like this is the only diff between what was generated - Does the `CreateGetMetricsRequest` method not do the right thing with a null content? I can't seem to find where it is implemented.

Reply:
Generated code and this implementation have the following differences. 
(1) Generated code function, did not have the default value of `RequestContent content = null`, it was required to pass the value for `content` this has been changed in the custom code and made the default value null. 
(2) As you commented here, we are checking if `content` is `null` and if so then we are passing the object. The reason being our service requires it to be blank. 

You can find `CreateGetMetricsRequest` here: https://github.com/NiveditJain/azure-sdk-for-net/blob/f106f064086cc51b1c540229ac56d205ed3c1cc9/sdk/loadtestservice/Azure.Developer.LoadTesting/src/Generated/LoadTestRunClient.cs#L914-L942

Reply:
Should the swagger be adjusted to make content an option parameter instead?

Reply:
@christothes The body is an optional parameter. But the backend of the service is implemented in .NET 6 and due to the limitation of .NET 6, `Content-Type` header needs to be passed even when no body is passed. Hence, we resorted to passing an empty JSON as body so that the header is passed always. Similar changes are implemented in Java SDK as well due to the above-mentioned reason. This should be resolved by upgrading the service to .NET 7, which would be in the near future. After that, we can revert these changes in SDKs. cc:// @NiveditJain